### ISSUE TITLE
Process with only one UO crashes during flowrate calculations

### ISSUE THREAD

**Issue by ronald-jaepel** [fau-advanced-separations/CADET-Process#90]
When using the MRE from [here ](https://github.com/modsim/CADET/issues/149), we get a crash when trying to simulate during the calculation of flow rates step, because there are no flow rates. This error is not raised by either of:
```
process.check_config()
process.check_cstr_volume()
```

here's the MRE:

```python
from CADETProcess.processModel import ComponentSystem
from CADETProcess.processModel import Cstr
from CADETProcess.processModel import FlowSheet
from CADETProcess.processModel import Process
from CADETProcess.simulator import Cadet

component_system = ComponentSystem(1)

reactor = Cstr(component_system, name='reactor')
reactor.porosity = 0.5
reactor.c = [1]
reactor.V = 1e-3
reactor.solution_recorder.write_solution_bulk = True

flow_sheet = FlowSheet(component_system)
flow_sheet.add_unit(reactor)

process = Process(flow_sheet, 'process')
process.cycle_time = 10

process.check_config()
process.check_cstr_volume()

simulator = Cadet()
sim_results = simulator.simulate(process)
```

with the error message 

```stdout
There was an exception in simulate_n_cycles
Traceback (most recent call last):
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\log.py", line 172, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\simulator.py", line 306, in simulate_n_cycles
    return self.run(process, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 136, in wrapper
    results = func(self, process, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 420, in run
    cadet.root = self.get_process_config(process)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 136, in wrapper
    results = func(self, process, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 510, in get_process_config
    config.input.model = self.get_input_model(process)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 778, in get_input_model
    input_model.connections = self.get_model_connections(process)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 800, in get_model_connections
    section_states = process.flow_rate_section_states
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\dataStructure\cache.py", line 13, in __get__
    value = super().__get__(instance, cls)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\processModel\process.py", line 215, in flow_rate_section_states
    for unit, unit_flow_rates in self.flow_rate_timelines.items():
                                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\dataStructure\cache.py", line 13, in __get__
    value = super().__get__(instance, cls)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\processModel\process.py", line 161, in flow_rate_timelines
    flow_rates = self.flow_sheet.get_flow_rates(state)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\processModel\flowSheet.py", line 634, in get_flow_rates
    coeffs = np.array(list(flow_rates.values()), ndmin=2)[:, i]
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
IndexError: index 0 is out of bounds for axis 1 with size 0
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2023.2.3\plugins\python-ce\helpers\pydev\pydevconsole.py", line 364, in runcode
    coro = func()
           ^^^^^^
  File "<input>", line 1, in <module>
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2023.2.3\plugins\python-ce\helpers\pydev\_pydev_bundle\pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2023.2.3\plugins\python-ce\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:\Users\ronal\PycharmProjects\CADET-Process\tmp\bulk_solution.py", line 22, in <module>
    sim_results = simulator.simulate(process)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\simulator.py", line 250, in simulate
    results = self.simulate_n_cycles(
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\log.py", line 150, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\log.py", line 201, in wrapper
    results = function(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\log.py", line 180, in wrapper
    raise e
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\log.py", line 172, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\simulator.py", line 306, in simulate_n_cycles
    return self.run(process, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 136, in wrapper
    results = func(self, process, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 420, in run
    cadet.root = self.get_process_config(process)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 136, in wrapper
    results = func(self, process, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 510, in get_process_config
    config.input.model = self.get_input_model(process)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 778, in get_input_model
    input_model.connections = self.get_model_connections(process)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\simulator\cadetAdapter.py", line 800, in get_model_connections
    section_states = process.flow_rate_section_states
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\dataStructure\cache.py", line 13, in __get__
    value = super().__get__(instance, cls)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\processModel\process.py", line 215, in flow_rate_section_states
    for unit, unit_flow_rates in self.flow_rate_timelines.items():
                                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\dataStructure\cache.py", line 13, in __get__
    value = super().__get__(instance, cls)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\processModel\process.py", line 161, in flow_rate_timelines
    flow_rates = self.flow_sheet.get_flow_rates(state)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ronal\PycharmProjects\CADET-Process\CADETProcess\processModel\flowSheet.py", line 634, in get_flow_rates
    coeffs = np.array(list(flow_rates.values()), ndmin=2)[:, i]
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
IndexError: index 0 is out of bounds for axis 1 with size 0
```

**Comment by schmoelder** [comment]
@daklauss

**Comment by schmoelder** [comment]
Thanks for bringing this up. This should be resolved in a7229c8
